### PR TITLE
THF-182: Adding Metatags for social media sharing.

### PR DESCRIPTION
**Pull request by xkhaven**
- Metatags for social media sharing

### CODE REVIEW

File: pages/[...slug].tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ import { getQueryParamsFor } from '@/lib/params'
 import { NavProps, FooterProps } from "@/lib/types"
 import { getBreadCrumb, getLanguageLinks } from '@/lib/helpers'
 import { useReactAndShare } from '@/hooks/useAnalytics'
+import {useTranslation} from "next-i18next";
```
Comment (Jussiles) [inline pages/[...slug].tsx:27]: Coding style nit picking: Fix the spacing here.

File: pages/[...slug].tsx [hunk 2]
```diff
@@ -116,20 +117,30 @@ export async function getStaticProps(context: GetStaticPropsContext): Promise<Ge
...
     return <ErrorPage statusCode={404} />
   }
 
   if (!node) return null
 
+  const { t } = useTranslation('common')
+  const metaTitle = node.field_metatags?.title ? node.field_metatags.title : node.title
+  const metaDescription = node.field_metatags?.description ? node.field_metatags.description : node.field_lead_in
+  const metaUrl = process.env.NEXT_PUBLIC_SITE_URL + router.asPath
+  const metaImage = node?.field_image_url ? node.field_image_url : process.env.NEXT_PUBLIC_SITE_URL + "/tyollisyyspalvelut.png"
```
Comment (Jussiles) [inline pages/[...slug].tsx:131]: Seems that only Event content type is using `field_image_url`. You could also add metaimage support for Landing page image that means getting the hero image if it's set on Landing page (usually landing page should use hero with image).

Add also metatags for /pages/index.js so the frontpage will have them also.

**Review by Jussiles — COMMENTED** [review]
Looks good! Added comments about things that could be added.

File: pages/[...slug].tsx (hunk 1 above)
Comment (xkhaven) [reply pages/[...slug].tsx]: Fixed.

File: pages/[...slug].tsx (hunk 2 above)
Comment (xkhaven) [reply pages/[...slug].tsx]: Added Meta info for index page. Updated image information per content type.

**Review by Jussiles — APPROVED** [review]
Looks good! 🦜 Couple of small coding style things that could be fixed before merging.

**Pull request merged**